app/services/accept_unconditional_offer.rb

```diff
@@ -0,0 +1,20 @@
+class AcceptUnconditionalOffer
+  def initialize(application_choice:)
+    @application_choice = application_choice
+  end
+
+  def save!
+    ActiveRecord::Base.transaction do
+      ApplicationStateChange.new(@application_choice).accept_unconditional_offer!
+      @application_choice.update!(accepted_at: Time.zone.now, recruited_at: Time.zone.now)
+    end
+
+    NotificationsList.for(@application_choice, include_ratifying_provider: true).each do |provider_user|
+      ProviderMailer.unconditional_offer_accepted(provider_user, @application_choice).deliver_later
+    end
+
+    CandidateMailer.unconditional_offer_accepted(@application_choice).deliver_later
+
+    StateChangeNotifier.new(:recruited, @application_choice).application_outcome_notification
```

Comment:
Not sure but is the state change notifier something we usually/can test for?

Reply:
:+1: https://github.com/DFE-Digital/apply-for-teacher-training/pull/4154/commits/45abd80da804b4b7ba26030e77c8686870e86214